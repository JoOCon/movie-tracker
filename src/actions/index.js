export const addMovies = (movieData) => ({
  type: 'ADD_MOVIES',
  movieData
});

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const setActiveUser = ({ id, name }) => ({
  type: 'SET_ACTIVE_USER',
  id,
  name
});

export const setUserFavorites = (favorites) => ({
  type: 'SET_USER_FAVORITES',
  favorites
});