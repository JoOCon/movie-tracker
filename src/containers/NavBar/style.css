.nav-bar {
  color: #ffcb9a;
  display: flex;
  align-items: center;
  justify-content: center; }

.login {
  margin-right: .5em;
  width: 5em;
  font-size: 1em;
  padding: .2em .5em;
  border-radius: 1em;
  border: 0px;
  background-color: #d9b08c;
  color: #d1e8e2; }

.signup {
  width: 5em;
  font-size: 1em;
  padding: .2em .5em;
  border-radius: 1em;
  border: 0px;
  background-color: #d9b08c;
  color: #d1e8e2; }
