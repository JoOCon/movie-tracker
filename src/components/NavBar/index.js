import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>Login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};
