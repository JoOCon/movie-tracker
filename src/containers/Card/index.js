import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import { addFavorite } from '../../actions';
import { postFavorites } from '../../api/apiCalls';
import './style.css';

export const Card = (props) => {
  const { movie, user, addFavorite } = props;
  const { poster_path, vote_average } = movie;
  const rating = vote_average / 2;

  const handleFavorite = () => {
    const favoriteData = {...movie, user_id: user.id};
    addFavorite(favoriteData);
    postFavorites(favoriteData);
  };

  return (
    <div className='card'>
      <div className='favorite-wrapper'>
        <button 
          className='favorite' 
          onClick={() => handleFavorite()}>
        </button>
        <StarRatings 
          starDimension={'1em'} 
          rating={rating} 
          numberOfStars={5} 
          starRatedColor={'gold'} 
        />
      </div>
      <img className='poster-image' src={poster_path} alt='movie data' />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userData
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  postFavorites: (movie) => dispatch(postFavorites(movie))
});

Card.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object,
  addFavorite: PropTypes.func,
  postFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
