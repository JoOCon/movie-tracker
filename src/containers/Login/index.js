import React, { Component } from 'react';
import './style.css';
import { login } from '../../api/apiCalls'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await login(this.state);
    // this.setState({ userName: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='email' placeholder='email' type='email' value={email} onChange={this.handleChange} />
        <input name='password' placeholder='password' type='password' value={password} onChange={this.handleChange} />
        <button>LogIn</button>
      </form>
    );
  }
}
