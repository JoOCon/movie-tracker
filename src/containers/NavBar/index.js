import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};