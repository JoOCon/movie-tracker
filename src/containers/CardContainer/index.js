import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import './style.css';

const CardContainer = (props) => {
  const displayMovies = props.movies.map(movie => <Card key={movie.id} movie={movie} />);
  return (
    <div className="card-container">
      {displayMovies}
    </div>
  );
};

const mapStateToProps = (state) => ({
  movies: state.moviesData
});

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps)(CardContainer);
