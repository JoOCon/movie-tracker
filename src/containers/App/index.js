import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { addMovies } from '../../actions';
import { fetchMovies } from '../../api/apiCalls';
import CardContainer from '../CardContainer';
import FavoriteContainer from '../FavoriteContainer';
import Login from '../../containers/Login';
import SignUp from '../../containers/SignUp';
import { Header } from '../../components/Header';
import './style.css';

export class App extends Component {
  componentDidMount = () => {
    this.fetchOpeningMovies();
  }

  fetchOpeningMovies = async () => {
    const { addMovies } = this.props;
    const openingMovies = await fetchMovies();
    addMovies(openingMovies);
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Route path='/' component={Header} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/signup' component={SignUp} />
          <Route exact path='/' component={CardContainer} />
          <Route exact path='/:landingPage' component={CardContainer} />
          <Route exact path='/:user/favorites' component={FavoriteContainer} />
        </div>
      </BrowserRouter>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (moviesData) => dispatch(addMovies(moviesData))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default connect(null, mapDispatchToProps)(App);
