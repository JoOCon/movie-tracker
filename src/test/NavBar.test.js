import React from 'react';
import { NavBar, mapStateToProps, mapDispatchToProps } from '../containers/NavBar';
import { shallow } from 'enzyme';
import * as mockData from '../test/mockData';
import * as actions from '../actions';

describe('NavBar tests', () => {
  let wrapper;
  let mockFunction;
  const { mockUserData } = mockData;
  beforeEach(() => {
    mockFunction = jest.fn();
    wrapper = shallow( 
      <NavBar 
        user={mockUserData}
        resetTheStore={mockFunction}
      /> 
    );
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    const { mockUserData } = mockData;
    it('should have a users data object in props', () => {
      const mockState = {userData: mockUserData};
      const expected = {user: mockUserData};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockDispatch;
    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should dispatch resetTheStore when called', () => {
      const { resetTheStore } = actions;
      const mockAction = resetTheStore();
      const props = mapDispatchToProps(mockDispatch);
      props.resetTheStore();
      expect(mockDispatch).toBeCalledWith(mockAction);
    });
  });
});
