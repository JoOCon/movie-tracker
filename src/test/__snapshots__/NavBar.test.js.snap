// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "name": "Taylor",
            "user_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          path="/:user"
/>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/:user",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/:user"
/>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/:user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
